File: back-end/TeusControle/Infrastructure/Dtos/SortingParams.cs
using TeusControle.Infrastructure.Enums;

namespace TeusControle.Infrastructure.Dtos
{
    public class SortingParams
    {
        /// <summary>
        /// Tipo de ordenação
        /// </summary>
        public SortEnum SortOrder { get; set; } = SortEnum.Asc;

        /// <summary>
        /// Nome da coluna
        /// </summary>
        public string ColumnName { get; set; }
    }
}

File: teus_controle_ui/lib/core/models/paginated/sorting_params.dart
import 'package:json_annotation/json_annotation.dart';

import 'enums/sort_enum.dart';

part 'sorting_params.g.dart';

@JsonSerializable()
class SortingParams {
  late final SortEnum sortOrder;
  late final String columnName;

  SortingParams({
    this.sortOrder = SortEnum.asc,
    required this.columnName,
  });

  factory SortingParams.fromJson(Map<String, dynamic> json) =>
      _$SortingParamsFromJson(json);

  Map<String, dynamic> toJson() => _$SortingParamsToJson(this);
}
